components/data_stores/actions/get-record-keys/get-record-keys.mjs

```diff
@@ -0,0 +1,133 @@
+import app from "../../data_stores.app.mjs";
+
+export default {
+  key: "data_stores-get-record-keys",
+  name: "Get Record Keys",
+  description: "Get all record keys in your [Pipedream Data Store](https://pipedream.com/data-stores/) that mathces with your query. The memory consumption of the workflow can be affected, since this action will be exposing, to the workflow, the entire data from the selected datastore",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    app,
+    dataStore: {
+      propDefinition: [
+        app,
+        "dataStore",
+      ],
+    },
+    query: {
+      type: "string",
+      label: "Query",
+      description: "The query to filter the records.",
+    },
+    contains: {
+      type: "boolean",
+      label: "Contains",
+      description: "If `true`, the query will be used to filter the records that contains the query. If `false`, the query will be used to filter the records that are equal to the query.",
+      optional: true,
+    },
+    caseInsensitive: {
+      type: "boolean",
+      label: "Case Insensitive",
+      description: "If `true`, the query will be used to filter the records based on case-insensitivity. If `false`, the query will be used to filter the records that are equal to the query.",
+      optional: true,
+    },
+  },
+  methods: {
+    getAllKeysWithRecord(
+      obj,
+      query,
+      contains = false,
+      caseInsensitive = false,
+      path = "",
+    ) {
+      const filteredKeys = [];
+      const values = [];
+
+      const helper = (obj, path) => {
+        // if the value is an array, iterate through it
+        if (Array.isArray(obj)) {
+          for (let i = 0; i < obj.length; i++) {
+            helper(obj[i], path + i + ".");
+          }
+          return;
+        }
+
+        // if the value is an object, iterate through it
+        else if (typeof obj === "object" && obj !== null) {
+          for (let key in obj) {
+            helper(obj[key], path + key + ".");
+          }
+          return;
+        }
+
+        // if the value is a string, check if it matches the query
+        if (!contains && !caseInsensitive && obj == query) {
+          filteredKeys.push(path.slice(0, -1));
+          values.push(obj);
+          return;
+        }
+
+        // `contains` and `caseInsensitive` are both true
+        if (contains && caseInsensitive && String(obj).toLowerCase()
+          .includes(query.toLowerCase())
+        ) {
+          filteredKeys.push(path.slice(0, -1));
+          values.push(obj);
+          return;
+        }
+
+        // `contains` is true and `caseInsensitive` is false
+        if (contains && !caseInsensitive && String(obj).includes(query)) {
+          filteredKeys.push(path.slice(0, -1));
+          values.push(obj);
+          return;
+        }
+
+        // `contains` is false and `caseInsensitive` is true
+        if (!contains && caseInsensitive
+          && String(obj).toLocaleLowerCase() == query.toLowerCase()
```

Comment:
Is there a reason for it to be `.toLocaleLowerCase()`?

Reply:
No reason, didn't notice that